job-server-api/src/spark.jobserver/NamedObjectSupport.scala

```diff
@@ -0,0 +1,134 @@
+package spark.jobserver
+
+import akka.util.Timeout
+import java.lang.ThreadLocal
+import java.util.concurrent.atomic.AtomicReference
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.storage.StorageLevel
+import scala.concurrent.duration.Duration
+
+sealed trait NamedObject
+case class NamedRDD[T](rdd: RDD[T], storageLevel: StorageLevel) extends NamedObject
+case class NamedDataFrame(df: DataFrame) extends NamedObject
+
+abstract class NamedObjectPersister[O <: NamedObject] {
+  //TODO what is the context needed for? c: ContextLike,
+  def saveToContext(namedObj: O, name: String): Unit
+  //TODO - what is this for?
+  //def getFromContext(c: ContextLike, name: String): O
```

Comment:
I did not quite understand what this method should be good for and left it therefor out for now.